developer/frameworks/articles/290-configuration/05-reading-configurations-configuration-provider.markdown

```diff
@@ -0,0 +1,115 @@
+    at the `PORTLET_INSTANCE` scope, but only works at that scope.
+
+## Using the Configuration Provider [](id=using-the-configuration-provider)
+
+When using the Configuration Provider, instead of receiving the configuration
+directly, the class that wants to access it must 
+
+1.  Receive a `ConfigurationProvider` to obtain the configuration.
+
+2.  Be registered with a `ConfigurationBeanDeclaration`.
+
+The tutorial on 
+[scoping configurations](/docs/7-2/frameworks/-/knowledge_base/f/scoping-configurations) 
+demonstrates how to register the configuration with a `ConfigurationBeanDeclaration`.
+
+After registering with a `ConfigurationBeanDeclaration`, you're ready to use a
+`ConfigurationProvider` to retrieve the scoped configuration. Here's how you
+obtain a reference to it:
+
+1.  Here's the approach for components:
+
+        @Referencomponents.e
+        protected void setConfigurationProvider(ConfigurationProvider configurationProvider) {
+            _configurationProvider = configurationProvider;
+        }
+
+2.  Here's the approach for Service Builder services:
+
+        @ServiceReference(type = ConfigurationProvider.class)
+        protected ConfigurationProvider configurationProvider;
+
+3.  For Spring beans, it is possible to use the same mechanism as for Service
+    Builder services (`@ServiceReference`). Check the documentation on 
+    [how to integrate Spring beans with OSGi services](/docs/7-2/frameworks/-/knowledge_base/f/invoking-services-from-service-builder-code#referencing-an-osgi-service)
+    for more details.
+
+4.  For anything else, call the same methods from the utility class,
+    `ConfigurationProviderUtil`. Be sure you call the utility methods in
+    contexts where the portal is guaranteed to be initialized prior to the
+    method call. This class is useful in the 
+    [scripting console](/docs/7-2/user/-/knowledge_base/u/running-scripts-from-the-script-console),
+    for example. Here's an example method that uses the utility class. It comes
+    from the export-import service, which is only called during the import and
+    export of content from a running portal:
+
+	protected boolean isValidateLayoutReferences() throws PortalException {
+		long companyId = CompanyThreadLocal.getCompanyId();
+
+		ExportImportServiceConfiguration exportImportServiceConfiguration =
+			ConfigurationProviderUtil.getCompanyConfiguration(
+				ExportImportServiceConfiguration.class, companyId);
+
+		return exportImportServiceConfiguration.validateLayoutReferences();
+	}
+
+To retrieve the configuration, use one of the following methods of the provider:
+
+`getCompanyConfiguration()`
+: Used when you want to support different configurations per virtual instance.
+In this case, the configuration is usually entered by an admin through Control
+Panel &rarr; Configuration &rarr; Instance Settings. Since this UI is not
+automatically generated (yet) you must extend the UI with your own form.
```

Comment:
This should be updated to indicate that the UI _can_ be auto-generated with proper configuration interface scoping.

Reply:
Whoops.